test/HeaderTest.hpp

```diff
@@ -0,0 +1,42 @@
+#include "catch.hpp"
```

Comment:
Are there tests around m_request_headers when it resizes? (with new header class maybe that's not necessary?)

Reply:
I think its unnecessary, but I will add one

Reply:
was definitely necessary